src/services/PwmService/PwmService.ts

```diff
@@ -0,0 +1,183 @@
+  }
+
+  /**
+   * @async Calls the PWM get_duty_cycle SDK method through an RPC request
+   */
+  async getDutyCycle(PWM: PWM): Promise<number> {
+    const response = await this.PWMRequest("get_duty_cycle", PWM, "EdgePiRPC_PWM.GetDutyCycle");
+    return (response.content as GetDutyCycle).dutyCycle;
+  }
+
+  /**
+   * @async Calls the PWM get_polarity SDK method through an RPC request
+   */
+  async getPolarity(PWM: PWM): Promise<Polarity> {
+    const response = await this.PWMRequest("get_polarity", PWM, "EdgePiRPC_PWM.GetPolarity");
+    return (response.content as GetPolarity).polarity;
+  }
+
+  /**
+   * @async Calls the PWM get_enabled SDK method through an RPC request
+   */
+  async getEnabled(PWM: PWM): Promise<boolean> {
+    const response = await this.PWMRequest("get_enabled", PWM, "EdgePiRPC_PWM.GetEnabled");
+    return (response.content as GetEnabled).enabled;
+  }
+
+  /**
+   * @async Calls the PWM set_config SDK method through RPC
+   */
+  async setConfig({
+    pwmNum,
+    frequency = undefined,
+    dutyCycle = undefined,
+    polarity = undefined,
+  }: PWMConfig): Promise<string> {
+    const requestType = this.serviceProtoRoot.lookupType(
+      "EdgePiRPC_PWM.Config"
+    );
+    const responseType = this.serviceProtoRoot.lookupType(
+      "EdgePiRPC_PWM.SuccessMsg"
+    );
+    const argsList = createConfigArgsList({ pwmNum, frequency, dutyCycle, polarity });
+
+    const response = await this.handlePWMRequest(
+      requestType,
+      responseType,
+      "set_config",
+      { confArg: argsList }
+    );
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Sends an RPC request for a specified PWM SDK method
+   */
+  private async PWMRequest(
+    methodName: string,
+    pwmNum: PWM,
+    responseTypeName: string,
+  ): Promise<serverResponse> {
+    const requestType = this.serviceProtoRoot.lookupType(
+      "EdgePiRPC_PWM.PWM"
+    );
+    const responseType = this.serviceProtoRoot.lookupType(
+      responseTypeName
+    );
+
+    return await this.handlePWMRequest(
+      requestType,
+      responseType,
+      methodName,
+      { pwmNum }
+    );
+  }
+
+  /**
+   * @async Handles an RPC request and returns a response message
+   */
+  private async handlePWMRequest(
+    requestType: protobuf.Type,
+    responseType: protobuf.Type,
+    methodName: string,
+    requestMsg: Object
+  ): Promise<serverResponse> {
+    const serviceReq: serviceRequest = {
+      serviceName: this.serviceName,
+      methodName,
+      requestMsg,
+    };
+
+    console.info(
+      `Calling PWM ${methodName.replace(/_([a-z])/g, (_, group1) =>
+        group1.toUpperCase()
+      )}` + ` through Rpc Channel.`
+    );
+
+    const response: serverResponse = await this.rpcChannel.callMethod(
+      serviceReq,
+      requestType,
+      responseType
+    );
+
+    if (response.error !== undefined) {
+      throw Error(response.error);
+    }
```

Comment:
Is it possible for response to be null after the call to `this.rpcChannel.callMethod`? If so, wouldn't checking `response.error` raise an error?

Reply:
You're right! I updated to check for that.